SAMtemplates/alarms/main.yaml

```diff
@@ -81,3 +88,175 @@ Resources:
+      ActionsEnabled: !Ref EnableAlerts
+      AlarmActions:
+        - !ImportValue lambda-resources:SlackAlertsSnsTopicArn
+      InsufficientDataActions:
+        - !ImportValue lambda-resources:SlackAlertsSnsTopicArn
+      OKActions:
+        - !ImportValue lambda-resources:SlackAlertsSnsTopicArn
+
+  PrescriptionStatusUpdateUnhandledErrorsAlarm:
+    Type: AWS::CloudWatch::Alarm
+    Properties:
+      AlarmDescription: Count of PrescriptionStatusUpdate unhandled errors
+      AlarmName: !Sub ${StackName}_PrescriptionStatusUpdate_UnhandledErrors
+      Namespace: Lambda
+      MetricName: Errors
+      Dimensions:
+        - Name: FunctionName
+          Value: !Ref UpdatePrescriptionStatusFunctionName
+      Period: 60 #seconds
+      EvaluationPeriods: 1
+      Statistic: Sum
+      ComparisonOperator: GreaterThanOrEqualToThreshold
+      Threshold: 1
+      Unit: Count
+      TreatMissingData: notBreaching
+      ActionsEnabled: !Ref EnableAlerts
+      AlarmActions:
+        - !ImportValue lambda-resources:SlackAlertsSnsTopicArn
+      InsufficientDataActions:
+        - !ImportValue lambda-resources:SlackAlertsSnsTopicArn
+      OKActions:
+        - !ImportValue lambda-resources:SlackAlertsSnsTopicArn
+
+  DynamoDBSystemErrorsAlarm:
+    Type: AWS::CloudWatch::Alarm
+    Properties:
+      AlarmDescription: Count of DynamoDB system errors
+      AlarmName: !Sub ${StackName}_DynamoDB_SystemErrors
+      Namespace: DynamoDB
+      MetricName: SystemErrors
+      Dimensions:
+        - Name: TableName
+          Value: !Ref PrescriptionStatusUpdatesTableName
+      Period: 60 #seconds
+      EvaluationPeriods: 1
+      Statistic: Sum
+      ComparisonOperator: GreaterThanOrEqualToThreshold
+      Threshold: 1
+      Unit: Count
+      TreatMissingData: notBreaching
+      ActionsEnabled: !Ref EnableAlerts
+      AlarmActions:
+        - !ImportValue lambda-resources:SlackAlertsSnsTopicArn
+      InsufficientDataActions:
+        - !ImportValue lambda-resources:SlackAlertsSnsTopicArn
+      OKActions:
+        - !ImportValue lambda-resources:SlackAlertsSnsTopicArn
+
+  CPSUFhirConverterErrorLogsMetricFilter:
+    Type: AWS::Logs::MetricFilter
+    Properties:
+      FilterName: CPSUFhirConverterErrors
+      FilterPattern: !Sub '{ ($.function_name = "${ConvertRequestToFhirFormatFunctionName}") && ($.response.statusCode = 500) }'
+      LogGroupName:
+        Fn::ImportValue: !Sub ${StackName}:functions:${ConvertRequestToFhirFormatFunctionName}:LambdaLogGroupName
+      MetricTransformations:
+        - MetricNamespace: LambdaLogFilterMetrics
+          MetricName: ErrorCount
+          MetricValue: 1
+          Unit: Count
+          Dimensions:
+            - Key: FunctionName
+              Value: $.function_name
+
+  CPSUFhirConverterErrorsAlarm:
+    Type: AWS::CloudWatch::Alarm
+    Properties:
+      AlarmDescription: Count of CPSU FHIR Converter errors
+      AlarmName: !Sub ${StackName}_CPSU_FHIR_Converter_Errors
```

Comment:
So i think the extra underscores in the alarm name will end up with the alarm name coming through to slack being a bit wrong, may need to concat the middle section so it ends up something like ${StackName}_CpsuFhirConverter_Errors

Reply:
27a3173c67787437bd27d7e821cbf8b84014a017